source/Coop/Mod/Persistence/Party/MovementData.cs

```diff
@@ -93,6 +93,20 @@ public bool NearlyEquals(MovementData other)
                    TargetSettlement?.Id == other.TargetSettlement?.Id;
         }
 
+        public override bool Equals(Object obj)
+        {
+            if (!(obj is MovementData))
+            {
+                return false;
+            }
+
+            MovementData other = (MovementData)obj;
+            return DefaultBehaviour.Equals(other.DefaultBehaviour) &&
+                   TargetPosition.Equals(other.TargetPosition) &&
```

Comment:
Do you really want to exactly compare the float coordinates? I think the position should be compared using `Compare.CoordinatesEqual(Vec2 a, Vec2 b)`. As map coordinates are compressed for synchronization (`Compression.Coordinate2d`), floating point errors are to be expected. I realize you're using the new Equals method to compare local values only, but for the sake of usability i think it would be best to implement it using the coordinate compare anyways. This might prevent comparision errors in the future. Or what do you think?

Looking at the class, there's already a `MovementData.NearlyEquals(MovementData other)` which should use `Compare.CoordinatesEqual(Vec2 a, Vec2 b)` as well. I think it would be good if you combined `Equals` and `NearlyEquals` into a single method, probably just `Equals`.

Reply:
You're right, I've changed the implementation but instead of using CoordinatesEqual I've used the NearlyEquals implemented by TaleWorlds in the Vec2 Class, using the precision in the Compare Class.

![image](https://user-images.githubusercontent.com/23215672/103768370-24f29880-5001-11eb-902a-166c56a288bc.png)